core/src/main/scala/pureconfig/ConfigCursor.scala

```diff
@@ -272,6 +383,9 @@ case class ConfigListCursor(value: ConfigList, pathElems: List[String], offset:
    */
   def list: List[ConfigCursor] =
     value.asScala.toList.drop(offset).zipWithIndex.map { case (cv, idx) => ConfigCursor(cv, indexKey(idx) :: pathElems) }
+
+  // Avoid resetting the offset when using ConfigCursor's implementation.
+  override def asListCursor: Either[ConfigReaderFailures, ConfigListCursor] = Right(this)
```

Comment:
Well spotted 👍 For efficiency reasons, can you override `asObjectCursor` on `ConfigObjectCursor`, too? Also, can you add a test covering this offset bug?

Reply:
Sure. I've overridden `asObjectCursor` in 7f37c88 and added a test covering the offset bug in 4823cag.